lib/tmdb_handler.rb

```diff
@@ -81,52 +82,72 @@ def tmdb_handler_add_movie(tmdb_id)
     popularity: @movie.popularity, runtime: @movie.runtime, mpaa_rating: @movie.mpaa_rating)
   end
 
-  def tmdb_handler_update_movie(tmdb_id)
-    @movie_url = "https://api.themoviedb.org/3/movie/#{tmdb_id}?api_key=#{ENV['tmdb_api_key']}&append_to_response=trailers,credits,similar,releases"
-    @result = JSON.parse(open(@movie_url).read, symbolize_names: true)
-    @movie_to_update = Movie.find_by(tmdb_id: tmdb_id)
-    @movie = MovieMore.tmdb_info(@result)
-
-    @movie_to_update.update_attributes(title: @movie.title, imdb_id: @movie.imdb_id, genres: @movie.genres,
-      actors: @movie.actors, adult: @result[:adult], backdrop_path: @movie.backdrop_path,
-      poster_path: @movie.poster_path, release_date: @movie.release_date, overview: @movie.overview,
-      trailer: @movie.trailer, director: @movie.director, director_id: @movie.director_id,
-      vote_average: @movie.vote_average, popularity: @movie.popularity, runtime: @movie.runtime,
-      mpaa_rating: @movie.mpaa_rating)
+  def self.tmdb_handler_update_movie(tmdb_id)
+    movie = Movie.find_by(tmdb_id: tmdb_id)
+    Raven.capture_message("movie not found in our db: #{tmdb_id}") && return unless movie
+
+    movie_url = "#{BASE_URL}/movie/#{tmdb_id}?api_key=#{ENV['tmdb_api_key']}&append_to_response=trailers,credits,similar,releases"
+    api_result = HTTParty.get(movie_url).deep_symbolize_keys rescue nil
+    Raven.capture_message("API request failed for tmdb_id: #{tmdb_id}") && return unless api_result && api_result[:id] == tmdb_id
+
+    updated_data = MovieMore.tmdb_info(api_result)
+
+    if movie.title != updated_data.title
+      msg = "Movie title doesn't match. tmdb_id: #{tmdb_id}. Current title: #{movie.title}. Updated title: #{updated_data.title}"
+      Raven.capture_message(msg) && return
+    end
+
+    movie.update!(
+      title: updated_data.title,
+      imdb_id: updated_data.imdb_id,
+      genres: updated_data.genres,
+      actors: updated_data.actors,
+      backdrop_path: updated_data.backdrop_path,
+      poster_path: updated_data.poster_path,
+      release_date: updated_data.release_date,
+      overview: updated_data.overview,
+      trailer: movie.trailer || updated_data.trailer,
+      director: updated_data.director,
+      director_id: updated_data.director_id,
+      vote_average: updated_data.vote_average,
+      popularity: updated_data.popularity,
+      runtime: updated_data.runtime,
+      mpaa_rating: updated_data.mpaa_rating
+    )
   end
 
   def tmdb_handler_actor_more(actor_id)
     tmdb_handler_person_detail_search(actor_id)
   end
 
   def tmdb_handler_person_detail_search(person_id)
-    api_bio_url = "https://api.themoviedb.org/3/person/#{person_id}?api_key=#{ENV['tmdb_api_key']}"
+    api_bio_url = "#{BASE_URL}/person/#{person_id}?api_key=#{ENV['tmdb_api_key']}"
```

Comment:
This is just a code cleanup so we're not repeating the same base url everywhere. So when it changes from `/3/` to `/4` or whatever, it only needs to be updated in one place.

Reply:
Yes. Very nice!